tests/animal_module_tests/test_animal_manager.py

```diff
@@ -425,9 +433,59 @@ def test_print_animal_num_warnings(animal_manager: AnimalManager):
         assert add_log.call_count == 2
 
 
-def test_init_nutrient_rqmts():
+def test_animals_by_type(mocker: MockerFixture):
+    """Unit test for function animals_by_type in file routines/animal/animal_manager.py"""
+    mocker.patch('RUFAS.routines.animal.animal_manager.AnimalManager.__init__', return_value=None)
+    mock_animal_manager = AnimalManager()
+    mock_animal_manager.calves = MagicMock()
+    mock_animal_manager.heiferIs = MagicMock()
+    mock_animal_manager.heiferIIs = MagicMock()
+    mock_animal_manager.heiferIIIs = MagicMock()
+    mock_animal_manager.cows = MagicMock()
+    actual = mock_animal_manager.animals_by_type
+    expected = {
+            Calf: mock_animal_manager.calves,
+            HeiferI: mock_animal_manager.heiferIs,
+            HeiferII: mock_animal_manager.heiferIIs,
+            HeiferIII: mock_animal_manager.heiferIIIs,
+            Cow: mock_animal_manager.cows,
+        }
+    assert actual == expected
+
+
+def test_init_nutrient_rqmts(mocker: MockerFixture) -> None:
     """Unit test for function init_nutrient_rqmts in file routines/animal/animal_manager.py"""
-    pass
+    mocker.patch('RUFAS.routines.animal.animal_manager.AnimalManager.__init__', return_value=None)
+    mock_animal_manager = AnimalManager()
+    mock_animal_manager.ANIMAL_GROUPING_SCENARIO = None
+    mock_animals = [MagicMock(), MagicMock()]
+    for animal in mock_animals:
+        animal.body_weight = 100
+        animal.calc_nutrient_rqmts = MagicMock()
+        animal.set_nutrient_rqmts = MagicMock()
+    animal_types = ['calves', 'heiferIs', 'heiferIIs', 'heiferIIIs', 'cows']
+    for animal_type in animal_types:
+        setattr(mock_animal_manager, animal_type, mock_animals)
+
+    mock_weather = MagicMock()
+    mock_current_conditions = MagicMock()
+    mock_current_conditions.mean_air_temperature = 27
+    mock_weather.get_current_day_conditions.return_value = mock_current_conditions
+    mock_time = MagicMock()
+    mock_feed = MagicMock()
+
+    mock_animal_manager.init_nutrient_rqmts(mock_weather, mock_time, mock_feed)
+
+    mock_weather.get_current_day_conditions.assert_called_once_with(mock_time)
+    assert 1 == animal.calc_nutrient_rqmts.call_count
+    assert 4 == animal.set_nutrient_rqmts.call_count
+    animal.calc_nutrient_rqmts.assert_called_with(mock_feed, 27)
+    animal.set_nutrient_rqmts.assert_has_calls([call(None), call(27, None)], any_order=True)
+    target_p_animal = 0.0072 * 100 * 1000
```

Comment:
```suggestion
    expected_p_animal = 720.0
```
These multiplications duplicate test logic, so I think it would be better to hardcode the expected value. Either way, it might be good to add more context to this value. 

Also, I think the term `expected_p_animal` would be a better name because (to me) it makes it clearer that this is a value related specifically to this unit test and not something related to the quality or validity of 720.0 being the value for p_animal.